### PR TITLE
server: asynchronous host resolving.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay

### CODE REVIEW

File: src/server/resolve.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    tx: Sender<Msg>,
+    h: Option<JoinHandle<()>>,
+}
+
+impl Resolver {
+    pub fn new() -> Result<Resolver> {
+        let (tx, rx) = mpsc::channel();
+        let builder = thread::Builder::new().name("host-resolver".to_owned());
+        let h = try!(builder.spawn(move || {
+            while let Ok(Msg::Peer { addr, cb }) = rx.recv() {
```
Comment (BusyJay) [inline src/server/resolve.rs]: Do we need to cache connection?

Comment (siddontang) [inline src/server/resolve.rs]: Server already caches it.

**Comment by BusyJay** [comment]
LGTM

File: src/server/mod.rs [hunk 2]
```diff
@@ -101,6 +103,12 @@ pub enum Msg {
         addr: String,
         data: ConnData,
     },
+    // Send data to remote peer with parsed socket address.
+    SendPeerSock {
+        sock: SocketAddr,
```
Comment (ngaut) [inline src/server/mod.rs]: sock and addr make me confused.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**